tests/unit_tests/test_server_client.py

```diff
@@ -405,6 +405,44 @@ async def mul_2(x: int) -> int:
             assert await composite_runnable_2.ainvoke(3) == 10
 
 
+@pytest.mark.asyncio
+async def test_config_updater(
+    event_loop: AbstractEventLoop, mocker: MockerFixture
+) -> None:
+    """Test updating the config based on the raw request object."""
+
+    async def add_one(x: int) -> int:
+        """Add one to simulate a valid function"""
+        return x + 1
+
+    app = FastAPI()
+
+    def config_updater(config: Dict[str, Any], request: Request) -> Dict[str, Any]:
+        """Update the config"""
+        config = config.copy()
+        if "metadata" in config:
+            config["metadata"] = config["metadata"].copy()
+        else:
+            config["metadata"] = {}
+        config["metadata"]["headers"] = request.headers
+        return config
+
+    server_runnable = RunnableLambda(add_one)
+
+    add_routes(app, server_runnable, path="/add_one", config_updater=config_updater)
```

Comment:
Maybe call this `config_extractor`?

And how do I actually call this as an end user? A `config` body param?

Reply:
This is for picking up information from request headers / cookies if needed for some reason and placing it inside the config.